src/protocols/secure_channel/SessionEstablishmentTransport.h

```diff
@@ -0,0 +1,72 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines Application Channel class. The object of this
+ *      class can be used by CHIP data model cluster applications to send
+ *      and receive messages. The messages are encrypted using session keys.
+ */
+
+#pragma once
+
+#include <messaging/ExchangeTransport.h>
+#include <transport/TransportMgr.h>
+
+namespace chip {
+namespace Transport {
+class BLE;
+}
+
+class SessionEstablishmentTransport : public Messaging::ExchangeTransport
```

Comment:
should we move SessionEstablishmentTransport to messing layer together with ApplicationExchangeTransport? It seems most of interactions of this clas occurs in exchange layer and it is used by ExchangeContext.

Reply:
My reason to keep it out of messaging layer was to prohibit some other code accidentally using it. It's structured as a delegate class whose implementation is provided by the user (PASESession in this scenario).

Reply:
acked